api/src/main/java/org/openmrs/module/patientqueueing/api/PatientQueueingService.java

```diff
@@ -26,6 +28,30 @@
  */
 public interface PatientQueueingService extends OpenmrsService {
 	
+	/**
+	 * This Method Gets a patientQueue By the queueNumber
+	 * 
+	 * @param queueNumber The queueNumber String that is used to search for a patientQueue
+	 * @return a patientQueue that matches the queueNumber passed
+	 * @throws APIException
+	 */
+	@Transactional
+	public PatientQueue getPatientQueueByQueueNumber(String queueNumber);
```

Comment:
What are you using this method for?

Reply:
@dkayiwa am using it to search for a patient queueNumber at the moment in private method isQueueNumberIdExisting